libraries/mempool/mempool.cpp

```diff
@@ -22,435 +20,535 @@ namespace koinos::mempool {
-      multi_index::ordered_unique< multi_index::tag< by_account >,
-         multi_index::member< account_resources_object, account_type, &account_resources_object::account >
-      >
-   >
->;
+   crypto::multihash node_id;
 
-class mempool_impl final
-{
-private:
-   account_resources_index          _account_resources_idx;
-   mutable std::mutex               _account_resources_mutex;
+   try
+   {
+      auto root_id = _db.get_root( lock )->id();
 
-   pending_transaction_queue        _pending_transactions;
-   pending_transaction_index        _pending_transaction_idx;
-   mutable std::mutex               _pending_transaction_mutex;
+      // We are handling the genesis case
+      if ( root_id != _db.get_head( lock )->id() )
+         node_id = tmp_id( previous_id );
+      else
+         node_id = tmp_id( root_id );
 
-public:
-   mempool_impl();
-   virtual ~mempool_impl();
+      auto node = _db.get_node( node_id, lock );
+      assert( node );
 
-   bool has_pending_transaction( const transaction_id_type& id ) const;
-   std::vector< rpc::mempool::pending_transaction > get_pending_transactions( std::size_t limit );
-   bool check_pending_account_resources(
-      const account_type& payer,
-      uint64_t max_payer_resources,
-      uint64_t trx_resource_limit ) const;
-   uint64_t add_pending_transaction(
-      const protocol::transaction& transaction,
-      std::chrono::system_clock::time_point time,
-      uint64_t max_payer_rc,
-      uint64_t disk_storaged_used,
-      uint64_t network_bandwidth_used,
-      uint64_t compute_bandwidth_used );
-   std::pair< uint64_t, uint64_t > remove_pending_transactions( const std::vector< transaction_id_type >& ids );
-   uint64_t prune( std::chrono::seconds expiration, std::chrono::system_clock::time_point now );
-   std::size_t payer_entries_size() const;
-   void cleanup_account_resources( const pending_transaction_object& pending_trx );
-   std::size_t pending_transaction_count() const;
+      std::vector< transaction_id_type > ids;
 
-private:
-   bool check_pending_account_resources_lockfree(
-         const account_type& payer,
-         uint64_t max_payer_rc,
-         uint64_t rc_limit
-      )const;
-};
+      for ( int i = 0; i < bam.block().transactions_size(); ++i )
+         ids.emplace_back( bam.block().transactions( i ).id() );
 
-mempool_impl::mempool_impl() {}
-mempool_impl::~mempool_impl() = default;
+      const auto removed_count = remove_pending_transactions( node, ids );
+
+      auto new_id = util::converter::to< crypto::multihash >( bam.block().id() );
+      _db.update_node( node_id, new_id, bam.block().header(), lock );
+      node_id = new_id;
+
+      if ( bam.live() && removed_count )
+         LOG(info) << "Removed " << removed_count << " included transaction(s) via block - Height: "
+            << bam.block().header().height() << ", ID: " << util::to_hex( bam.block().id() );
+   }
+   catch ( ... )
+   {
+      _db.discard_node( node_id, lock );
```

Comment:
With the `assert( false )` I am assuming we never expect to hit this?

Can we add a `LOG(error)` since the discarding of the temp node will probably cause problems as the mempool will desync from chain?